packages/@aws-cdk/aws-events/lib/input.ts

```diff
@@ -148,7 +148,7 @@ class FieldAwareEventInput extends RuleTargetInput {
       }
 
       public resolveToken(t: Token, _context: IResolveContext) {
-        if (!isEventField(t)) { return t; }
+        if (!isEventField(t)) { return Token.asString(t); }
```

Comment:
Yeah, it used to inherit a working `toString()`. Implementing that on the Token would also have worked. I think I would feel safer if we did that... but that can wait until we uncover a new failing case maybe.

Reply:
Would it have worked with token encoded as numbers?